prow/plugins/label/label_test.go

```diff
@@ -0,0 +1,181 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package label
+
+import (
+	"fmt"
+	"testing"
+
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/github/fakegithub"
+	"reflect"
+)
+
+const (
+	fakeRepoOrg  = "fakeOrg"
+	fakeRepoName = "fakeName"
+	orgMember    = "Alice"
+	nonOrgMember = "Bob"
+	prNumber     = 1
+)
+
+var (
+	existingLabels = []string{"area/infra", "area/api", "priority/critical", "priority/urgent", "priority/important"}
+)
+
+func formatLabels(labels ...string) []string {
+	r := []string{}
+	for _, l := range labels {
+		r = append(r, fmt.Sprintf("%s/%s#%d:%s", fakeRepoOrg, fakeRepoName, prNumber, l))
+	}
+	if len(r) == 0 {
+		return nil
+	}
+	return r
+}
+
+func getFakeRepo(commentBody, commenter string, initial_labels []string) (*fakegithub.FakeClient, github.IssueCommentEvent) {
+	fc := &fakegithub.FakeClient{
+		IssueComments:  make(map[int][]github.IssueComment),
+		OrgMembers:     []string{orgMember},
+	}
+	startingLabels := []github.Label{}
+	for _, l := range initial_labels {
+		startingLabels = append(startingLabels, github.Label{Name: l})
+	}
+
+	ice := github.IssueCommentEvent{
+		Repo: github.Repo{
+			Owner: github.User{Name: fakeRepoOrg},
+			Name:  fakeRepoName,
+		},
+		Comment: github.IssueComment{
+			Body: commentBody,
+			User: github.User{Login: commenter},
+		},
+		Issue: github.Issue{
+			User:        github.User{Login: "a"},
+			Number:      prNumber,
+			PullRequest: &struct{}{},
+			Labels:      startingLabels,
+		},
+	}
+	return fc, ice
+}
+
+func TestLabel(t *testing.T) {
+	// "a" is the author, "a", "r1", and "r2" are reviewers.
```

Comment:
@apelisse @spxtr I need help figuring out a new testing strategy.  Before I'd compare fc.AddedLabels to expectedLabels but since I'm using fc.AddedLabels as the list of all labels (see changes to fakeClient where I added GetLabels func) I can no longer make that comparison.  Can I modify the functionality of handle so that it returns the labels that are successfully added to the issue?